src/compiler/checker.ts

```diff
@@ -14478,107 +14479,29 @@ namespace ts {
         // Here, performing a full type check of the body of the function expression whilst in the process of
         // determining the type of foo would cause foo to be given type any because of the recursive reference.
         // Delaying the type check of the body ensures foo has been assigned a type.

```

Comment:
Move the comment up to the definition of `deferredCheckNodes` and reword the comment appropriately.